mlem/contrib/requirements.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+from typing import ClassVar, Optional
+
+from mlem.core.objects import MlemBuilder, MlemModel
+from mlem.core.requirements import Requirements
+from mlem.ui import EMOJI_OK, EMOJI_PACK, echo
+
+REQUIREMENTS = "requirements.txt"
+
+logger = logging.getLogger(__name__)
+
+
+class RequirementsBuilder(MlemBuilder):
+    """MlemBuilder implementation for building requirements"""
+
+    type: ClassVar = "requirements"
+
+    target: Optional[str] = None
+    """Target path for the requirements.txt file"""
+    platform: str = "pip"
+    """Type of requirements, example: unix"""
+
+    def write_requirements_file(self, requirements: Requirements):
+        requirement_string = "\n".join(requirements.to_pip())
+        if self.target is None:
+            print(requirement_string)
+        else:
+            echo(EMOJI_PACK + "Generating requirements file...")
+            with open(os.path.join(self.target), "w", encoding="utf8") as req:
+                logger.debug(
+                    "Auto-determined requirements for model: %s.",
+                    requirements.to_pip(),
+                )
+                req.write(requirement_string + "\n")
+                echo(EMOJI_OK + f"{self.target} generated!")
+
+    def build(self, obj: MlemModel):
+        if self.platform == "unix":
```

Comment:
Use `load_impl_ext("requirement", self.platform)` to get python type and then use `.of_type` method to get the list of requirements of this type. If target is `None`, print them, otherwise write them to file.
This way it will work for any requirement type

Reply:
You may also want to add abstract method to `Requirement` to get "installable" representation of those so you can `" ".join(r.get_repr() for r in reqs)`

Reply:
done, `custom` requirements is left I guess, the last comment of yours basically.

Reply:
hmm, one does need a identifier on when to materialize vs. when to print and a check on that identifier is basically an if/else thing so we can't really avoid it I guess. Do you have any suggestions @mike0sv ?